Fix:

error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
warning: ISO C does not allow extra ‘;’ outside of a function [-Wpedantic]
error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
warning: implicit declaration of function ‘strncasecmp’; did you mean ‘strncmp’? [-Wimplicit-function-declaration]

https://pagure.io/xmlto/c/1375e2df75530cd198bd16ac3de38e2b0d126276
https://pagure.io/xmlto/c/412f97cdc00d2bbf7e8121012b49fc07b3fe3d2b
https://pagure.io/xmlto/c/8e34f087bf410bcc5fe445933d6ad9bae54f24b5
https://pagure.io/xmlto/c/6fa6a0e07644f20abf2596f78a60112713e11cbe
--- xmlif/xmlif.l.orig	2015-02-11 06:20:51.000000000 -0600
+++ xmlif/xmlif.l	2024-04-04 04:16:13.000000000 -0500
@@ -37,13 +37,14 @@
  */
 #include <string.h>
 #include <stdlib.h>
+#include <strings.h>
 
 #define TRUE	1
 #define FALSE	0
 
 static char **selections;	/* selection tokens */
 static int nselections;		/* number of selections */
-static ifsense;			/* sense of last `if' or unless seen */
+static int ifsense;		/* sense of last `if' or unless seen */
 static char *attribute;		/* last attribute scanned */
 
 struct stack_t {
@@ -237,9 +238,9 @@
 
 #include "config.h"
 
-int yywrap() {exit(0);};
+int yywrap() {exit(0);}
 
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
     int i;
 
@@ -265,7 +266,7 @@
 	    exit(1);
 	}
 
-    yylex();
+    return yylex();
 }
 
 /*
